runtime/compiler/runtime/JITClientSession.cpp

```diff
@@ -786,20 +788,44 @@ ClientSessionHT::purgeOldDataIfNeeded()
    {
    PORT_ACCESS_FROM_PORT(TR::Compiler->portLib);
    int64_t crtTime = j9time_current_time_millis();
+   bool incomplete;
+   bool lowMemory = false;
+   uint64_t freePhysicalMemory = compInfo->computeAndCacheFreePhysicalMemory(incomplete, 20); //check if memory is free
+   if (freePhysicalMemory != OMRPORT_MEMINFO_NOT_AVAILABLE && !incomplete)
+      {
+      if (freePhysicalMemory < TR::Options::getSafeReservePhysicalMemoryValue())
+         {
+         lowMemory = true;
+         }
+      }
```

Comment:
This code should be moved under the following `if` statement (where `lowMemory` is used) so that we don't execute it for nothing.